{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Intermine-Python: Tutorial 5: Query Results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This tutorial will talk about dealing with the results of our query. You can either store the results into a file (using a library like csv) or you can process the results immediately after you extract them. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will write a short query and will then explore the Results class of intermine. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from intermine.webservice import Service"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "service = Service(\"https://www.flymine.org/flymine/service\")\n",
    "query=service.new_query(\"Gene\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "<intermine.query.Query at 0x11276bf70>"
     },
     "metadata": {},
     "execution_count": 3
    }
   ],
   "source": [
    "query.select(\"publications.*\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "<TernaryConstraint: Gene LOOKUP zen IN D. melanogaster>"
     },
     "metadata": {},
     "execution_count": 4
    }
   ],
   "source": [
    "query.add_constraint(\"Gene\",\"LOOKUP\",\"zen\",extra_value=\"D. melanogaster\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once we have added our constraints and views, we are ready to look at the results. The results can be accessed in either a dictionary form, or a list, a ResultRow object(the most common one), or even as a list of strings (CSV or TSV)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "tags": [
     "outputPrepend",
     "outputPrepend",
     "outputPrepend",
     "outputPrepend"
    ]
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": " the absence of eve, and so repression is not likely to be the result of eve competitively inhibiting an activator protein from binding to the same DNA element. Instead, we propose that eve protein is probably interfering with the function of proteins bound at other locations in the promoter. The biochemical demonstration that a Drosophila homeodomain protein can directly regulate RNA synthesis strengthens the view that this class of regulators act as transcription factors to control development.' publications.doi='10.1016/0092-8674(89)90424-8' publications.firstAuthor='Biggin M D' publications.id=1007999 publications.issue='3' publications.journal='Cell' publications.month='Aug' publications.pages='433-40' publications.pubMedId='2569362' publications.title='A purified Drosophila homeodomain protein represses transcription in vitro.' publications.volume='58' publications.year=1989\nGene: publications.abstractText=\"mRNA-associated processes and gene structure in eukaryotes are typically treated as separate research subjects. Here, we bridge this separation and leverage the extensive multidisciplinary work on Drosophila melanogaster to examine the roles that capping, splicing, cleavage/polyadenylation, and telescripting (i.e, the protection of nascent transcripts from premature cleavage/polyadenylation by the splicing factor U1) might play in shaping exon-intron architecture in protein-coding genes. Our findings suggest that the distance between subsequent internal 5' splice sites (5'ss) in Drosophila genes is constrained such that telescripting effects are maximized, in theory, and thus nascent transcripts are less vulnerable to premature termination. Exceptionally weak 5'ss and constraints on intron-exon size at the gene 5' end also indicate that capping might enhance the recruitment of U1 and, in turn, promote telescripting at this location. Finally, a positive correlation between last exon length and last 5'ss strength suggests that optimal donor splice sites in the proximity of the pre-mRNA tail may inhibit the processing of downstream polyadenylation signals more than weak donor splice sites do. These findings corroborate and build upon previous experimental and computational studies on Drosophila genes. They support the possibility, hitherto scantly explored, that mRNA-associated processes impose significant constraints on the evolution of eukaryotic gene structure.\" publications.doi='10.1534/g3.116.029231' publications.firstAuthor='Lepennetier Gildas' publications.id=1000167 publications.issue='6' publications.journal='G3 (Bethesda)' publications.month='06' publications.pages='1617-26' publications.pubMedId='27172210' publications.title='mRNA-Associated Processes and Their Influence on Exon-Intron Structure in Drosophila melanogaster.' publications.volume='6' publications.year=2016\nGene: publications.abstractText=\"twist is one of the earliest expressed zygotically active genes required for dorsal-ventral pattern formation in the Drosophila embryo. Genetic studies suggest that this gene is activated in the ventral part of the blastoderm by maternally expressed dorsal gene product. Using P-element-mediated germ-line transformation, we have mapped a small (260 bp) dorsal-dependent ventral activator region (VAR) in the 5'-flanking region of the twist promoter that can direct the early ventral expression of a heterologous promoter. The VAR contains binding sites for a number of proteins present in extracts of Drosophila embryos. One of these sites bears homology to known binding sites for the dorsal transcription factor and is specifically bound by bacterially expressed dorsal protein. Furthermore, a 37-bp deletion that removes the dorsal-binding sequences abolishes the ventral-specific activity of the twist promoter constructs. Our data also show that additional sequences within the VAR are required to render the dorsal-binding sites functional. Finally, reverse genetic and biochemical data suggest that the transcription factor, encoded by the zeste gene may help control the overall level, if not the pattern, of twist expression.\" publications.doi='10.1101/gad.5.10.1892' publications.firstAuthor='Pan D J' publications.id=1008025 publications.issue='10' publications.journal='Genes Dev.' publications.month='Oct' publications.pages='1892-901' publications.pubMedId='1655573' publications.title='Functional analysis of the Drosophila twist promoter reveals a dorsal-binding ventral activator region.' publications.volume='5' publications.year=1991\nGene: publications.abstractText='wimp is a dominant maternal-effect mutation that interacts with a specific subset of early-acting maternal and zygotic Drosophila genes. We show that wimp is a change-of-function mutation, allelic to mutations of the 140-kD subunit of RNA polymerase, which causes reduced transcription of interacting genes. Loci that do not interact with wimp are expressed at normal levels. We discuss these results in terms of specific interactions between transcription factors and RNA polymerase. Embryos from wimp mothers show unaltered fate maps and develop normally, despite the reduction of transcript levels at least twofold. We suggest that spatial cues are determined by a balance of segmentation gene products rather than their absolute concentrations. We also demonstrate powerful genetic screens for otherwise undetected loci required for segmentation, sex determination, and other early functions.' publications.doi='10.1101/gad.5.3.341' publications.firstAuthor='Parkhurst S M' publications.id=1007912 publications.issue='3' publications.journal='Genes Dev.' publications.month='Mar' publications.pages='341-57' publications.pubMedId='2001838' publications.title='wimp, a dominant maternal-effect mutation, reduces transcription of a specific subset of segmentation genes in Drosophila.' publications.volume='5' publications.year=1991\nGene: publications.abstractText='zerknüllt (zen) is one of approximately 10 zygotically active genes that control the differentiation of the dorsal-ventral (D/V) pattern during early embryogenesis in Drosophila. Past genetic analyses suggest that maternal factors repress the expression of zen in ventral regions, thereby restricting zen products to dorsal and dorsal-lateral regions of precellular embryos. Subsequent interactions with other zygotic D/V regulatory genes refine the zen pattern, restricting expression to the dorsal-most ectoderm. Here we describe the use of zen promoter fusions and P-element transformation to identify cis elements that are responsible for the complex spatial pattern of zen expression. The zen promoter shows a two-tier organization: Distal sequences mediate its initial response to maternal factors, whereas proximal sequences are responsible for the refinement of the pattern in older embryos. The distal regulatory element has the property of a silencer (or anti-enhancer) element and can act over a distance to repress ventral expression of a heterologous promoter. Also, we discuss evidence that proximal promoter sequences interact with factors that may be modulated by a cell-cell communication pathway.' publications.doi='10.1101/gad.3.10.1518' publications.firstAuthor='Doyle H J' publications.id=1008117 publications.issue='10' publications.journal='Genes Dev.' publications.month='Oct' publications.pages='1518-33' publications.pubMedId='2612903' publications.title='Spatial regulation of zerknüllt: a dorsal-ventral patterning gene in Drosophila.' publications.volume='3' publications.year=1989\nGene: publications.abstractText='zerknüllt (zen) is unique among the 18 known homeo box genes in Drosophila since it is required for the differentiation of the dorsal-ventral pattern, and does not appear to be involved in the process of segmentation. Here we show that the zen region of the Antennapedia complex (ANT-C) consists of two closely linked homeo box genes, designated z1 and z2. The z1 and z2 transcription units show essentially identical patterns of expression during early development, which are consistent with the timing and sites of zen+ gene activity. The putative proteins encoded by z1 and z2 are highly divergent and are related only by virtue of homeo box homology. We have used P-element-mediated germ line transformation to show that z1 alone can provide zen+ gene function, suggesting that the z2 gene might be dispensable. The occurrence of closely linked homeo box genes that display similar patterns of expression is not unique to the zen locus. Such gene duplications might provide important clues to the evolution of the homeo box gene family in Drosophila and other organisms.' publications.doi='10.1101/gad.1.10.1268' publications.firstAuthor='Rushlow C' publications.id=1008001 publications.issue='10' publications.journal='Genes Dev.' publications.month='Dec' publications.pages='1268-79' publications.pubMedId='2892759' publications.title='Molecular characterization of the zerknüllt region of the Antennapedia gene complex in Drosophila.' publications.volume='1' publications.year=1987\nGene: publications.abstractText='β-Arrestins have been implicated in the regulation of multiple signalling pathways. However, their role in organism development is not well understood. In this study, we report a new in vivo function of the Drosophila β-arrestin Kurtz (Krz) in the regulation of two distinct developmental signalling modules: MAPK ERK and NF-κB, which transmit signals from the activated receptor tyrosine kinases (RTKs) and the Toll receptor, respectively. Analysis of the expression of effectors and target genes of Toll and the RTK Torso in krz maternal mutants reveals that Krz limits the activity of both pathways in the early embryo. Protein interaction studies suggest a previously uncharacterized mechanism for ERK inhibition: Krz can directly bind and sequester an inactive form of ERK, thus preventing its activation by the upstream kinase, MEK. A simultaneous dysregulation of different signalling systems in krz mutants results in an abnormal patterning of the embryo and severe developmental defects. Our findings uncover a new in vivo function of β-arrestins and present a new mechanism of ERK inhibition by the Drosophila β-arrestin Krz.' publications.doi='10.1038/emboj.2010.202' publications.firstAuthor='Tipping Marla' publications.id=1008090 publications.issue='19' publications.journal='EMBO J.' publications.month='Oct' publications.pages='3222-35' publications.pubMedId='20802461' publications.title='β-arrestin Kurtz inhibits MAPK and Toll signalling in Drosophila development.' publications.volume='29' publications.year=2010\nGene: publications.abstractText=None publications.doi='10.1002/(SICI)1521-1878(200003)22:3<305::AID-BIES13>3.0.CO;2-R' publications.firstAuthor='Wilkins A S' publications.id=1008035 publications.issue='3' publications.journal='Bioessays' publications.month='Mar' publications.pages='305-7' publications.pubMedId='10684591' publications.title='Drosophila: flying high in Zürich.' publications.volume='22' publications.year=2000\nGene: publications.abstractText=None publications.doi='10.1002/bies.950121107' publications.firstAuthor='White R J' publications.id=1008080 publications.issue='11' publications.journal='Bioessays' publications.month='Nov' publications.pages='537-9' publications.pubMedId='1982208' publications.title='Cell type-specific enhancement in the Drosophila embryo by consensus homeodomain binding sites.' publications.volume='12' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/0092-8674(90)90492-w' publications.firstAuthor='Hayashi S' publications.id=1008114 publications.issue='5' publications.journal='Cell' publications.month='Nov' publications.pages='883-94' publications.pubMedId='1979524' publications.title='What determines the specificity of action of Drosophila homeodomain proteins?' publications.volume='63' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/0092-8674(92)90471-n' publications.firstAuthor='McGinnis W' publications.id=1008048 publications.issue='2' publications.journal='Cell' publications.month='Jan' publications.pages='283-302' publications.pubMedId='1346368' publications.title='Homeobox genes and axial patterning.' publications.volume='68' publications.year=1992\nGene: publications.abstractText=None publications.doi='10.1016/0167-4781(94)00234-t' publications.firstAuthor='Courey A J' publications.id=1007932 publications.issue='1' publications.journal='Biochim. Biophys. Acta' publications.month='Mar' publications.pages='1-18' publications.pubMedId='7893745' publications.title='The establishment and interpretation of transcription factor gradients in the Drosophila embryo.' publications.volume='1261' publications.year=1995\nGene: publications.abstractText=None publications.doi='10.1016/0168-9525(90)90001-m' publications.firstAuthor='Laufer E' publications.id=1008005 publications.issue='8' publications.journal='Trends Genet.' publications.month='Aug' publications.pages='261-3' publications.pubMedId='7940752' publications.title='Evolution in developmental biology: of morphology and molecules.' publications.volume='10' publications.year=1994\nGene: publications.abstractText=None publications.doi='10.1016/0168-9525(91)90273-s' publications.firstAuthor='Budd P S' publications.id=1007930 publications.issue='3' publications.journal='Trends Genet.' publications.month='Mar' publications.pages='74-6' publications.pubMedId='1674389' publications.title='What do the regulators regulate? First glimpse downstream.' publications.volume='7' publications.year=1991\nGene: publications.abstractText=None publications.doi='10.1016/0304-419x(92)90008-m' publications.firstAuthor='Read D' publications.id=1008102 publications.issue='2-3' publications.journal='Biochim. Biophys. Acta' publications.month='Dec' publications.pages='79-93' publications.pubMedId='1457465' publications.title='Transcriptional regulators of Drosophila embryogenesis.' publications.volume='1114' publications.year=1992\nGene: publications.abstractText=None publications.doi='10.1016/s0065-2660(08)60028-0' publications.firstAuthor='Rushlow C' publications.id=1008015 publications.issue=None publications.journal='Adv. Genet.' publications.month=None publications.pages='277-307' publications.pubMedId='2112301' publications.title='Role of the zerknüllt gene in dorsal-ventral pattern formation in Drosophila.' publications.volume='27' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/s0065-2660(08)60029-2' publications.firstAuthor='Kaufman T C' publications.id=1005666 publications.issue=None publications.journal='Adv. Genet.' publications.month=None publications.pages='309-62' publications.pubMedId='1971986' publications.title='Molecular and genetic organization of the antennapedia gene complex of Drosophila melanogaster.' publications.volume='27' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/s0092-8674(00)80126-9' publications.firstAuthor='Miklos G L' publications.id=1008063 publications.issue='4' publications.journal='Cell' publications.month='Aug' publications.pages='521-9' publications.pubMedId='8752207' publications.title='The role of the genome project in determining gene function: insights from model organisms.' publications.volume='86' publications.year=1996\nGene: publications.abstractText=None publications.doi='10.1016/s0092-8674(00)80250-0' publications.firstAuthor='Bier E' publications.id=1007974 publications.issue='5' publications.journal='Cell' publications.month='May' publications.pages='681-4' publications.pubMedId='9182755' publications.title='Anti-neural-inhibition: a conserved mechanism for neural induction.' publications.volume='89' publications.year=1997\nGene: publications.abstractText=None publications.doi='10.1016/s0092-8674(00)81682-7' publications.firstAuthor='Fraser S E' publications.id=1007887 publications.issue='1' publications.journal='Cell' publications.month='Jan' publications.pages='41-55' publications.pubMedId='10647930' publications.title='The molecular metamorphosis of experimental embryology.' publications.volume='100' publications.year=2000\nGene: publications.abstractText=None publications.doi='10.1016/s0960-9822(01)00007-0' publications.firstAuthor='Brown S' publications.id=1007898 publications.issue='2' publications.journal='Curr. Biol.' publications.month='Jan' publications.pages='R43-4' publications.pubMedId='11231138' publications.title='A strategy for mapping bicoid on the phylogenetic tree.' publications.volume='11' publications.year=2001\nGene: publications.abstractText=None publications.doi='10.1038/35056022' publications.firstAuthor='Casci T' publications.id=1007955 publications.issue='3' publications.journal='Nat. Rev. Genet.' publications.month='Mar' publications.pages='161' publications.pubMedId='11256063' publications.title='Patterning back to front.' publications.volume='2' publications.year=2001\nGene: publications.abstractText=None publications.doi='10.1073/pnas.032685999' publications.firstAuthor='Michelson Alan M' publications.id=1008074 publications.issue='2' publications.journal='Proc. Natl. Acad. Sci. U.S.A.' publications.month='Jan' publications.pages='546-8' publications.pubMedId='11805309' publications.title='Deciphering genetic regulatory codes: a challenge for functional genomics.' publications.volume='99' publications.year=2002\nGene: publications.abstractText=None publications.doi='10.1093/emboj/17.23.6769' publications.firstAuthor='Shilo B Z' publications.id=1008115 publications.issue='23' publications.journal='EMBO J.' publications.month='Dec' publications.pages='6769-71' publications.pubMedId='9835652' publications.title='Flies over Crete: Drosophila molecular biology. Kolymbari, Crete, July 12-18, 1998.' publications.volume='17' publications.year=1998\nGene: publications.abstractText=None publications.doi='10.1101/gad.939601' publications.firstAuthor='Courey A J' publications.id=1008067 publications.issue='21' publications.journal='Genes Dev.' publications.month='Nov' publications.pages='2786-96' publications.pubMedId='11691830' publications.title='Transcriptional repression: the long and the short of it.' publications.volume='15' publications.year=2001\nGene: publications.abstractText=None publications.doi='10.1101/gad.976502' publications.firstAuthor='Barolo Scott' publications.id=1007905 publications.issue='10' publications.journal='Genes Dev.' publications.month='May' publications.pages='1167-81' publications.pubMedId='12023297' publications.title='Three habits of highly effective signaling pathways: principles of transcriptional control by developmental cell signaling.' publications.volume='16' publications.year=2002\nGene: publications.abstractText=None publications.doi='10.1146/annurev.bi.63.070194.002415' publications.firstAuthor='Gehring W J' publications.id=1007914 publications.issue=None publications.journal='Annu. Rev. Biochem.' publications.month=None publications.pages='487-526' publications.pubMedId='7979246' publications.title='Homeodomain proteins.' publications.volume='63' publications.year=1994\nGene: publications.abstractText=None publications.doi=None publications.firstAuthor='Denell R' publications.id=1007953 publications.issue='3' publications.journal='Genetics' publications.month='Nov' publications.pages='549-52' publications.pubMedId='7851753' publications.title='Discovery and genetic definition of the Drosophila Antennapedia complex.' publications.volume='138' publications.year=1994\nGene: publications.abstractText=None publications.doi=None publications.firstAuthor='Zhou J' publications.id=1008069 publications.issue=None publications.journal='Cold Spring Harb. Symp. Quant. Biol.' publications.month=None publications.pages='307-12' publications.pubMedId='9598364' publications.title='The regulation of enhancer-promoter interactions in the Drosophila embryo.' publications.volume='62' publications.year=1997\n"
    }
   ],
   "source": [
    "for gene in query.results(row=\"rr\"):\n",
    "    print(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "tags": [
     "outputPrepend",
     "outputPrepend",
     "outputPrepend"
    ]
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": " the absence of eve, and so repression is not likely to be the result of eve competitively inhibiting an activator protein from binding to the same DNA element. Instead, we propose that eve protein is probably interfering with the function of proteins bound at other locations in the promoter. The biochemical demonstration that a Drosophila homeodomain protein can directly regulate RNA synthesis strengthens the view that this class of regulators act as transcription factors to control development.' publications.doi='10.1016/0092-8674(89)90424-8' publications.firstAuthor='Biggin M D' publications.id=1007999 publications.issue='3' publications.journal='Cell' publications.month='Aug' publications.pages='433-40' publications.pubMedId='2569362' publications.title='A purified Drosophila homeodomain protein represses transcription in vitro.' publications.volume='58' publications.year=1989\nGene: publications.abstractText=\"mRNA-associated processes and gene structure in eukaryotes are typically treated as separate research subjects. Here, we bridge this separation and leverage the extensive multidisciplinary work on Drosophila melanogaster to examine the roles that capping, splicing, cleavage/polyadenylation, and telescripting (i.e, the protection of nascent transcripts from premature cleavage/polyadenylation by the splicing factor U1) might play in shaping exon-intron architecture in protein-coding genes. Our findings suggest that the distance between subsequent internal 5' splice sites (5'ss) in Drosophila genes is constrained such that telescripting effects are maximized, in theory, and thus nascent transcripts are less vulnerable to premature termination. Exceptionally weak 5'ss and constraints on intron-exon size at the gene 5' end also indicate that capping might enhance the recruitment of U1 and, in turn, promote telescripting at this location. Finally, a positive correlation between last exon length and last 5'ss strength suggests that optimal donor splice sites in the proximity of the pre-mRNA tail may inhibit the processing of downstream polyadenylation signals more than weak donor splice sites do. These findings corroborate and build upon previous experimental and computational studies on Drosophila genes. They support the possibility, hitherto scantly explored, that mRNA-associated processes impose significant constraints on the evolution of eukaryotic gene structure.\" publications.doi='10.1534/g3.116.029231' publications.firstAuthor='Lepennetier Gildas' publications.id=1000167 publications.issue='6' publications.journal='G3 (Bethesda)' publications.month='06' publications.pages='1617-26' publications.pubMedId='27172210' publications.title='mRNA-Associated Processes and Their Influence on Exon-Intron Structure in Drosophila melanogaster.' publications.volume='6' publications.year=2016\nGene: publications.abstractText=\"twist is one of the earliest expressed zygotically active genes required for dorsal-ventral pattern formation in the Drosophila embryo. Genetic studies suggest that this gene is activated in the ventral part of the blastoderm by maternally expressed dorsal gene product. Using P-element-mediated germ-line transformation, we have mapped a small (260 bp) dorsal-dependent ventral activator region (VAR) in the 5'-flanking region of the twist promoter that can direct the early ventral expression of a heterologous promoter. The VAR contains binding sites for a number of proteins present in extracts of Drosophila embryos. One of these sites bears homology to known binding sites for the dorsal transcription factor and is specifically bound by bacterially expressed dorsal protein. Furthermore, a 37-bp deletion that removes the dorsal-binding sequences abolishes the ventral-specific activity of the twist promoter constructs. Our data also show that additional sequences within the VAR are required to render the dorsal-binding sites functional. Finally, reverse genetic and biochemical data suggest that the transcription factor, encoded by the zeste gene may help control the overall level, if not the pattern, of twist expression.\" publications.doi='10.1101/gad.5.10.1892' publications.firstAuthor='Pan D J' publications.id=1008025 publications.issue='10' publications.journal='Genes Dev.' publications.month='Oct' publications.pages='1892-901' publications.pubMedId='1655573' publications.title='Functional analysis of the Drosophila twist promoter reveals a dorsal-binding ventral activator region.' publications.volume='5' publications.year=1991\nGene: publications.abstractText='wimp is a dominant maternal-effect mutation that interacts with a specific subset of early-acting maternal and zygotic Drosophila genes. We show that wimp is a change-of-function mutation, allelic to mutations of the 140-kD subunit of RNA polymerase, which causes reduced transcription of interacting genes. Loci that do not interact with wimp are expressed at normal levels. We discuss these results in terms of specific interactions between transcription factors and RNA polymerase. Embryos from wimp mothers show unaltered fate maps and develop normally, despite the reduction of transcript levels at least twofold. We suggest that spatial cues are determined by a balance of segmentation gene products rather than their absolute concentrations. We also demonstrate powerful genetic screens for otherwise undetected loci required for segmentation, sex determination, and other early functions.' publications.doi='10.1101/gad.5.3.341' publications.firstAuthor='Parkhurst S M' publications.id=1007912 publications.issue='3' publications.journal='Genes Dev.' publications.month='Mar' publications.pages='341-57' publications.pubMedId='2001838' publications.title='wimp, a dominant maternal-effect mutation, reduces transcription of a specific subset of segmentation genes in Drosophila.' publications.volume='5' publications.year=1991\nGene: publications.abstractText='zerknüllt (zen) is one of approximately 10 zygotically active genes that control the differentiation of the dorsal-ventral (D/V) pattern during early embryogenesis in Drosophila. Past genetic analyses suggest that maternal factors repress the expression of zen in ventral regions, thereby restricting zen products to dorsal and dorsal-lateral regions of precellular embryos. Subsequent interactions with other zygotic D/V regulatory genes refine the zen pattern, restricting expression to the dorsal-most ectoderm. Here we describe the use of zen promoter fusions and P-element transformation to identify cis elements that are responsible for the complex spatial pattern of zen expression. The zen promoter shows a two-tier organization: Distal sequences mediate its initial response to maternal factors, whereas proximal sequences are responsible for the refinement of the pattern in older embryos. The distal regulatory element has the property of a silencer (or anti-enhancer) element and can act over a distance to repress ventral expression of a heterologous promoter. Also, we discuss evidence that proximal promoter sequences interact with factors that may be modulated by a cell-cell communication pathway.' publications.doi='10.1101/gad.3.10.1518' publications.firstAuthor='Doyle H J' publications.id=1008117 publications.issue='10' publications.journal='Genes Dev.' publications.month='Oct' publications.pages='1518-33' publications.pubMedId='2612903' publications.title='Spatial regulation of zerknüllt: a dorsal-ventral patterning gene in Drosophila.' publications.volume='3' publications.year=1989\nGene: publications.abstractText='zerknüllt (zen) is unique among the 18 known homeo box genes in Drosophila since it is required for the differentiation of the dorsal-ventral pattern, and does not appear to be involved in the process of segmentation. Here we show that the zen region of the Antennapedia complex (ANT-C) consists of two closely linked homeo box genes, designated z1 and z2. The z1 and z2 transcription units show essentially identical patterns of expression during early development, which are consistent with the timing and sites of zen+ gene activity. The putative proteins encoded by z1 and z2 are highly divergent and are related only by virtue of homeo box homology. We have used P-element-mediated germ line transformation to show that z1 alone can provide zen+ gene function, suggesting that the z2 gene might be dispensable. The occurrence of closely linked homeo box genes that display similar patterns of expression is not unique to the zen locus. Such gene duplications might provide important clues to the evolution of the homeo box gene family in Drosophila and other organisms.' publications.doi='10.1101/gad.1.10.1268' publications.firstAuthor='Rushlow C' publications.id=1008001 publications.issue='10' publications.journal='Genes Dev.' publications.month='Dec' publications.pages='1268-79' publications.pubMedId='2892759' publications.title='Molecular characterization of the zerknüllt region of the Antennapedia gene complex in Drosophila.' publications.volume='1' publications.year=1987\nGene: publications.abstractText='β-Arrestins have been implicated in the regulation of multiple signalling pathways. However, their role in organism development is not well understood. In this study, we report a new in vivo function of the Drosophila β-arrestin Kurtz (Krz) in the regulation of two distinct developmental signalling modules: MAPK ERK and NF-κB, which transmit signals from the activated receptor tyrosine kinases (RTKs) and the Toll receptor, respectively. Analysis of the expression of effectors and target genes of Toll and the RTK Torso in krz maternal mutants reveals that Krz limits the activity of both pathways in the early embryo. Protein interaction studies suggest a previously uncharacterized mechanism for ERK inhibition: Krz can directly bind and sequester an inactive form of ERK, thus preventing its activation by the upstream kinase, MEK. A simultaneous dysregulation of different signalling systems in krz mutants results in an abnormal patterning of the embryo and severe developmental defects. Our findings uncover a new in vivo function of β-arrestins and present a new mechanism of ERK inhibition by the Drosophila β-arrestin Krz.' publications.doi='10.1038/emboj.2010.202' publications.firstAuthor='Tipping Marla' publications.id=1008090 publications.issue='19' publications.journal='EMBO J.' publications.month='Oct' publications.pages='3222-35' publications.pubMedId='20802461' publications.title='β-arrestin Kurtz inhibits MAPK and Toll signalling in Drosophila development.' publications.volume='29' publications.year=2010\nGene: publications.abstractText=None publications.doi='10.1002/(SICI)1521-1878(200003)22:3<305::AID-BIES13>3.0.CO;2-R' publications.firstAuthor='Wilkins A S' publications.id=1008035 publications.issue='3' publications.journal='Bioessays' publications.month='Mar' publications.pages='305-7' publications.pubMedId='10684591' publications.title='Drosophila: flying high in Zürich.' publications.volume='22' publications.year=2000\nGene: publications.abstractText=None publications.doi='10.1002/bies.950121107' publications.firstAuthor='White R J' publications.id=1008080 publications.issue='11' publications.journal='Bioessays' publications.month='Nov' publications.pages='537-9' publications.pubMedId='1982208' publications.title='Cell type-specific enhancement in the Drosophila embryo by consensus homeodomain binding sites.' publications.volume='12' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/0092-8674(90)90492-w' publications.firstAuthor='Hayashi S' publications.id=1008114 publications.issue='5' publications.journal='Cell' publications.month='Nov' publications.pages='883-94' publications.pubMedId='1979524' publications.title='What determines the specificity of action of Drosophila homeodomain proteins?' publications.volume='63' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/0092-8674(92)90471-n' publications.firstAuthor='McGinnis W' publications.id=1008048 publications.issue='2' publications.journal='Cell' publications.month='Jan' publications.pages='283-302' publications.pubMedId='1346368' publications.title='Homeobox genes and axial patterning.' publications.volume='68' publications.year=1992\nGene: publications.abstractText=None publications.doi='10.1016/0167-4781(94)00234-t' publications.firstAuthor='Courey A J' publications.id=1007932 publications.issue='1' publications.journal='Biochim. Biophys. Acta' publications.month='Mar' publications.pages='1-18' publications.pubMedId='7893745' publications.title='The establishment and interpretation of transcription factor gradients in the Drosophila embryo.' publications.volume='1261' publications.year=1995\nGene: publications.abstractText=None publications.doi='10.1016/0168-9525(90)90001-m' publications.firstAuthor='Laufer E' publications.id=1008005 publications.issue='8' publications.journal='Trends Genet.' publications.month='Aug' publications.pages='261-3' publications.pubMedId='7940752' publications.title='Evolution in developmental biology: of morphology and molecules.' publications.volume='10' publications.year=1994\nGene: publications.abstractText=None publications.doi='10.1016/0168-9525(91)90273-s' publications.firstAuthor='Budd P S' publications.id=1007930 publications.issue='3' publications.journal='Trends Genet.' publications.month='Mar' publications.pages='74-6' publications.pubMedId='1674389' publications.title='What do the regulators regulate? First glimpse downstream.' publications.volume='7' publications.year=1991\nGene: publications.abstractText=None publications.doi='10.1016/0304-419x(92)90008-m' publications.firstAuthor='Read D' publications.id=1008102 publications.issue='2-3' publications.journal='Biochim. Biophys. Acta' publications.month='Dec' publications.pages='79-93' publications.pubMedId='1457465' publications.title='Transcriptional regulators of Drosophila embryogenesis.' publications.volume='1114' publications.year=1992\nGene: publications.abstractText=None publications.doi='10.1016/s0065-2660(08)60028-0' publications.firstAuthor='Rushlow C' publications.id=1008015 publications.issue=None publications.journal='Adv. Genet.' publications.month=None publications.pages='277-307' publications.pubMedId='2112301' publications.title='Role of the zerknüllt gene in dorsal-ventral pattern formation in Drosophila.' publications.volume='27' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/s0065-2660(08)60029-2' publications.firstAuthor='Kaufman T C' publications.id=1005666 publications.issue=None publications.journal='Adv. Genet.' publications.month=None publications.pages='309-62' publications.pubMedId='1971986' publications.title='Molecular and genetic organization of the antennapedia gene complex of Drosophila melanogaster.' publications.volume='27' publications.year=1990\nGene: publications.abstractText=None publications.doi='10.1016/s0092-8674(00)80126-9' publications.firstAuthor='Miklos G L' publications.id=1008063 publications.issue='4' publications.journal='Cell' publications.month='Aug' publications.pages='521-9' publications.pubMedId='8752207' publications.title='The role of the genome project in determining gene function: insights from model organisms.' publications.volume='86' publications.year=1996\nGene: publications.abstractText=None publications.doi='10.1016/s0092-8674(00)80250-0' publications.firstAuthor='Bier E' publications.id=1007974 publications.issue='5' publications.journal='Cell' publications.month='May' publications.pages='681-4' publications.pubMedId='9182755' publications.title='Anti-neural-inhibition: a conserved mechanism for neural induction.' publications.volume='89' publications.year=1997\nGene: publications.abstractText=None publications.doi='10.1016/s0092-8674(00)81682-7' publications.firstAuthor='Fraser S E' publications.id=1007887 publications.issue='1' publications.journal='Cell' publications.month='Jan' publications.pages='41-55' publications.pubMedId='10647930' publications.title='The molecular metamorphosis of experimental embryology.' publications.volume='100' publications.year=2000\nGene: publications.abstractText=None publications.doi='10.1016/s0960-9822(01)00007-0' publications.firstAuthor='Brown S' publications.id=1007898 publications.issue='2' publications.journal='Curr. Biol.' publications.month='Jan' publications.pages='R43-4' publications.pubMedId='11231138' publications.title='A strategy for mapping bicoid on the phylogenetic tree.' publications.volume='11' publications.year=2001\nGene: publications.abstractText=None publications.doi='10.1038/35056022' publications.firstAuthor='Casci T' publications.id=1007955 publications.issue='3' publications.journal='Nat. Rev. Genet.' publications.month='Mar' publications.pages='161' publications.pubMedId='11256063' publications.title='Patterning back to front.' publications.volume='2' publications.year=2001\nGene: publications.abstractText=None publications.doi='10.1073/pnas.032685999' publications.firstAuthor='Michelson Alan M' publications.id=1008074 publications.issue='2' publications.journal='Proc. Natl. Acad. Sci. U.S.A.' publications.month='Jan' publications.pages='546-8' publications.pubMedId='11805309' publications.title='Deciphering genetic regulatory codes: a challenge for functional genomics.' publications.volume='99' publications.year=2002\nGene: publications.abstractText=None publications.doi='10.1093/emboj/17.23.6769' publications.firstAuthor='Shilo B Z' publications.id=1008115 publications.issue='23' publications.journal='EMBO J.' publications.month='Dec' publications.pages='6769-71' publications.pubMedId='9835652' publications.title='Flies over Crete: Drosophila molecular biology. Kolymbari, Crete, July 12-18, 1998.' publications.volume='17' publications.year=1998\nGene: publications.abstractText=None publications.doi='10.1101/gad.939601' publications.firstAuthor='Courey A J' publications.id=1008067 publications.issue='21' publications.journal='Genes Dev.' publications.month='Nov' publications.pages='2786-96' publications.pubMedId='11691830' publications.title='Transcriptional repression: the long and the short of it.' publications.volume='15' publications.year=2001\nGene: publications.abstractText=None publications.doi='10.1101/gad.976502' publications.firstAuthor='Barolo Scott' publications.id=1007905 publications.issue='10' publications.journal='Genes Dev.' publications.month='May' publications.pages='1167-81' publications.pubMedId='12023297' publications.title='Three habits of highly effective signaling pathways: principles of transcriptional control by developmental cell signaling.' publications.volume='16' publications.year=2002\nGene: publications.abstractText=None publications.doi='10.1146/annurev.bi.63.070194.002415' publications.firstAuthor='Gehring W J' publications.id=1007914 publications.issue=None publications.journal='Annu. Rev. Biochem.' publications.month=None publications.pages='487-526' publications.pubMedId='7979246' publications.title='Homeodomain proteins.' publications.volume='63' publications.year=1994\nGene: publications.abstractText=None publications.doi=None publications.firstAuthor='Denell R' publications.id=1007953 publications.issue='3' publications.journal='Genetics' publications.month='Nov' publications.pages='549-52' publications.pubMedId='7851753' publications.title='Discovery and genetic definition of the Drosophila Antennapedia complex.' publications.volume='138' publications.year=1994\nGene: publications.abstractText=None publications.doi=None publications.firstAuthor='Zhou J' publications.id=1008069 publications.issue=None publications.journal='Cold Spring Harb. Symp. Quant. Biol.' publications.month=None publications.pages='307-12' publications.pubMedId='9598364' publications.title='The regulation of enhancer-promoter interactions in the Drosophila embryo.' publications.volume='62' publications.year=1997\n"
    }
   ],
   "source": [
    "for gene in query.rows():\n",
    "    print(gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Iterating through query.results(row=\"rr\") and Iterating through query.rows() are equivalent. Feel free to use whichever you feel more comfortable with. If you want to extract only specific columns, it may be easier to use \"list\" instead of rr. Let's say you want to extract column 2 & 3, i.e. publications.doi and publications.firstAuthor, then it can be done as follows. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "10.1128/mcb.11.10.4909 Yamaguchi M\n10.1016/s0378-1119(03)00493-1 Kwon Eunjeong\n10.1016/j.yexcr.2013.10.010 Mannervik Mattias\n10.1006/jtbi.1996.0328 Bodnar J W\n10.1016/0092-8674(91)90651-e Ip Y T\n10.1101/gad.5.10.1881 Jiang J\n10.1038/332858a0 Hoey T\n10.1038/35101500 Gurdon J B\n10.1016/j.cub.2005.03.022 Schmidt-Ott Urs\n10.1242/dev.001065 Duboule Denis\n10.1016/s0959-437x(98)80036-1 Wray G A\n10.1002/1521-1878(200102)23:2<125::AID-BIES1019>3.0.CO;2-C Jagla K\n10.1002/bies.950170904 Venkatesh T V\n10.1073/pnas.97.9.4442 Patel N H\n10.1111/j.1096-0031.1996.tb00193.x Agosti D\n10.1016/0092-8674(92)90595-4 Geisler R\n10.1016/s1534-5807(03)00338-1 Hashimoto Carl\n10.1016/s0960-9822(95)00112-6 Jones C M\n10.1016/0092-8674(92)90596-5 Kidd S\nNone Carroll S B\n10.1007/s11060-016-2326-3 Jeibmann Astrid\n10.1242/dev.067926 Wilson Megan J\n10.1038/35068578 Ross J J\n10.1016/j.tig.2003.10.009 Raftery Laurel A\nNone Jaźwińska A\nNone González-Reyes A\n10.1371/journal.pgen.1002769 Holmqvist Per-Henrik\n10.1016/s0378-1119(01)00485-1 Nepveu A\n10.1101/gad.215459.113 Saunders Abbie\n10.1016/0012-1606(92)90254-e Strecker T R\n10.1093/gbe/evr061 Gehring Walter J\n10.1007/s00438-006-0187-8 Roy Swarnava\n10.1038/ncomms8102 Ugrankar Rupali\n10.1016/s0959-437x(02)00345-3 Markstein Michele\n10.1101/gr.104018.109 Ozdemir Anil\n10.1371/journal.pbio.1000456 Kazemian Majid\nNone Comeron J M\n10.1016/j.celrep.2019.03.071 Shokri Leila\n10.1101/gr.178701 Bergman C M\n10.1016/s0959-437x(02)00314-3 Averof Michalis\nNone Akam M\n10.1371/journal.pcbi.1001020 Su Jing\n10.1038/ng1908 Shirangi Troy R\n10.1080/19420889.2018.1447743 Anreiter Ina\n10.1074/jbc.271.7.3930 Hirose F\n10.1016/j.semcdb.2014.04.036 Hamaratoglu Fisun\n10.1016/j.devcel.2011.05.009 Kim Yoosik\nNone Affolter M\nNone Jiménez G\n10.7554/eLife.04837 Nitta Kazuhiro R\n10.1242/dev.019802 Pope Karen L\n10.1128/mcb.18.11.6584 Valentine S A\n10.1038/nature07214 Wang Xiaomeng\n10.1101/gad.11.15.1963 Huang A M\n10.1016/s0959-437x(96)80062-1 Rusch J\n10.1016/s0960-9822(95)00001-7 Ip Y T\n10.1016/j.ydbio.2014.02.007 Künnapuu Jaana\n10.1242/dev.00161 Chang Andy J\n10.1242/dev.00548 Reim Ingolf\nNone Manak J R\nNone Yip M L\n10.1073/pnas.1707467114 Poe Amy R\n10.1073/pnas.1713930115 Papagianni Aikaterini\n10.1126/science.aam5339 Zenk Fides\n10.1006/dbio.1994.1306 Warrior R\n10.1534/genetics.116.186783 Sandler Jeremy E\nNone Edgar B A\n10.1371/journal.pbio.0050117 De Renzis Stefano\n10.1038/sj.emboj.7601532 Astigarraga Sergio\n10.1016/0012-1606(84)90182-9 Wakimoto B T\n10.1016/0959-437x(93)90090-c Steward R\nNone Kumar Sudhir\n10.1006/jtbi.1995.0175 Takeda Y\n10.1101/gad.6.9.1654 Norris J L\n10.1101/gr.6490707 Copley Richard R\n10.1146/annurev-ento-120811-153601 Heffer Alison\n10.1016/s0965-1748(02)00089-9 Arnosti D N\n10.1093/nar/28.2.454 Decoville M\n10.1016/s0959-437x(03)00017-0 Hsia Cheryl C\n10.1038/380037a0 De Robertis E M\n10.1371/journal.pgen.1006868 Ribeiro Lupis\n10.1146/annurev.genet.35.102401.090832 Lall S\n10.1101/gad.1509607 Zeitlinger Julia\n10.1038/ncomms10115 Zwarts Liesbeth\n10.1016/0955-0674(93)90017-k Jäckle H\n10.1242/dev.01722 Xu Mu\n10.1371/journal.pone.0030610 Turki-Judeh Wiam\n10.1101/gad.188052.112 Kvon Evgeny Z\n10.1006/dbio.2002.0652 Stathopoulos Angelike\n10.1007/s00412-015-0543-8 Bürglin Thomas R\n10.1101/gr.209486.116 Koenecke Nina\n10.1016/j.ydbio.2006.08.027 Friedrich Markus\n10.1002/1097-010X(20001215)288:4<345::AID-JEZ7>3.0.CO;2-Y Coulier F\n10.1098/rstb.1995.0119 Akam M\n10.1101/gr.3468605 Negre Bárbara\n10.1016/0168-9525(94)90132-5 Duboule D\n10.1371/journal.pbio.0060027 Li Xiao-yong\n10.1038/323076a0 Doyle H J\n10.1016/j.cub.2010.01.040 Bothma Jacques P\nNone Stein D\n10.1002/bies.20285 McGregor Alistair P\n10.1101/gad.11.15.1949 Treisman J E\nNone Jiang J\nNone Rusch J\n10.1534/genetics.112.144915 Staller Max V\n10.1371/journal.pcbi.1004159 O'Connell Michael D\n10.1007/s00438-010-0591-y Roy Swarnava\n10.1038/nature07388 Liang Hsiao-Lan\n10.1038/nature03318 Wang Yu-Chiun\n10.1371/journal.pgen.1002086 Tsurumi Amy\n10.1371/journal.pgen.1002339 Nien Chung-Yi\n10.1016/j.ymeth.2014.01.003 Wunderlich Zeba\nNone Chen G\n10.1016/s1097-2765(03)00500-8 Senger Kate\n10.1016/s0955-0674(96)80010-x Gray S\n10.1186/1471-2148-6-106 Casillas Sònia\n10.1016/s0955-0674(96)80067-6 Treisman R\nNone Pan D\n10.1016/s0959-437x(00)00130-1 Tautz D\n10.1093/nar/gkr672 Mrinal Nirotpal\n10.7554/eLife.00861 Chen Kai\nNone Thisse C\n10.1016/j.ydbio.2010.06.026 Harrison Melissa M\n10.1042/bj3310001 Ogbourne S\n10.1534/genetics.109.110122 Salzer Claire L\n10.1016/s0968-0004(00)01597-8 Kortschak R D\n10.1073/pnas.012591199 Markstein Michele\n10.1016/j.gde.2004.07.004 Stathopoulos Angelike\n10.1186/1471-2105-5-202 Gurunathan Rajalakshmi\n10.1006/dbio.1997.8689 Kourakis M J\n10.1242/dev.02689 Lin Meng-chi\n10.1242/dev.02251 Pilot Fanny\n10.1016/j.cois.2016.01.009 Schmidt-Ott Urs\n10.1016/s0960-9822(03)00576-1 Farge Emmanuel\n10.1242/dev.02238 Ashe Hilary L\n10.1101/gad.494808 Miles Wayne O\n10.1093/nar/gkp619 Papatsenko Dmitri\n10.1038/nrg1724 Peel Andrew D\n10.1016/S0022-2836(05)80201-3 Hui C C\n10.1007/BF00375954 Jürgens Gerd\nNone Cribbs D L\n10.1371/journal.pgen.1005969 Kwon So Yeon\n10.1186/1752-0509-6-31 Kim Man-Sun\n10.1371/journal.pone.0001115 Aerts Stein\n10.1093/emboj/19.17.4463 Frémion F\n10.1038/nrg947 Simpson Pat\n10.1038/nature15545 Stampfel Gerald\n10.1016/s0012-1606(03)00353-1 Rudel David\n10.1016/0959-437x(93)90092-4 Hoch M\n10.1242/dev.079772 Liang Hsiao-Lan\n10.1101/gad.8.21.2602 Francois V\n10.1016/s0960-9822(06)00104-7 Ip Y T\n10.1016/j.devcel.2011.12.007 Reeves Gregory T\n10.1016/j.ydbio.2013.02.010 Duncan Elizabeth J\n10.1093/molbev/msu286 Moyers Bryan A\n10.1371/journal.pgen.1000178 Kwong Camilla\n10.1007/s10709-006-0031-4 Demuth Jeffery P\nNone Decotto E\nNone Long A D\n10.1038/nmeth.1548 Chung Kwanghun\n10.1016/j.cub.2003.12.036 Martinho Rui Gonçalo\nNone Kamens J\n10.1371/journal.pgen.1002905 Comeron Josep M\n10.1016/j.ygeno.2008.09.006 Bai Yongsheng\n10.1093/molbev/msw132 Quijano Janine C\n10.1016/s0959-437x(02)00344-1 Mann Richard S\nNone Ferguson E L\n10.1093/emboj/20.13.3298 Affolter M\nNone Nicholls R E\n10.1091/mbc.e06-09-0790 Ullah Zakir\n10.1101/gad.1350705 Massagué Joan\n10.1126/science.284.5414.606 Mannervik M\n10.1371/journal.pgen.1007631 Combs Peter A\n10.1038/nature08799 Schnorrer Frank\n10.1016/s0960-9822(03)00472-x Lynch Jeremy\n10.1093/nar/25.4.794 Ryu J R\n10.1101/gad.11.22.2952 Dubnicoff T\n10.1128/mcb.22.14.5089-5099.2002 Jia Songtao\n10.1101/gad.9.24.3177 Huang J D\n10.1016/s0378-1119(00)00161-x Chen G\n10.1038/sj.onc.1203223 Govind S\n10.1016/j.cub.2006.05.050 Chen Li-Ying\n10.1016/s0960-9822(99)80381-9 Dearden P\n10.1016/0955-0674(91)90112-c Hoffmann F M\nNone Jaynes J B\n10.1016/s0012-1606(03)00448-2 Chang Ting\n10.1073/pnas.96.7.3786 Stauber M\n10.1016/0925-4773(95)00439-8 Wimmer E A\n10.7554/eLife.12068 Sarov Mihail\n10.1073/pnas.89.17.7861 Govind S\nNone Johnson Hamlet M R\n10.1371/journal.pone.0098585 Heimiller Joseph\n10.1242/dev.02373 ten Bosch John R\n10.1534/genetics.166.3.1323 Yu Kweon\n10.1534/genetics.105.041327 Armstrong Jennifer A\n10.1016/0012-1606(89)90143-7 Schulz R A\nNone Winick J\n10.1002/bies.10180 Damen Wim G M\n10.1093/emboj/18.3.605 Asha H\n10.1128/mcb.15.7.3627 Tatei K\n10.1242/dev.02643 Ratnaparkhi Girish S\n10.1016/j.ymeth.2014.03.016 Peterson Aidan J\n10.1016/j.smim.2014.05.003 Igaki Tatsushi\nNone Hudson J B\n10.7554/eLife.34594 Liu Qinwen\n10.1073/pnas.93.18.9322 Kirchhamer C V\n10.1016/j.tig.2006.12.001 Negre Bárbara\n10.1073/pnas.90.1.143 Schubert F R\n10.1002/bies.950140302 Treisman J\nNone Randazzo F M\nNone Jiang J\n10.1101/gad.8.11.1247 Rusch J\nNone Kirov N\nNone Goff D J\n10.1146/annurev.ge.29.120195.002103 Morisato D\n10.1371/journal.pgen.1002266 Harrison Melissa M\n10.1101/gad.861401 Rushlow C\nNone Roth S\nNone Lewis R A\nNone Hirose F\n10.1073/pnas.97.9.4438 Akam M\n10.1371/journal.pone.0010820 Hueber Stefanie D\n10.1007/pl00006577 Schmid K J\n10.1126/science.287.5461.2185 Adams M D\nNone Häcker U\n10.1016/j.jtbi.2008.01.027 Ishihara Shuji\n10.1371/journal.pone.0029172 Garcia Mayra\n10.1128/mcb.10.3.872 Yamaguchi M\n10.1073/pnas.0408031102 Levine Michael\n10.1016/j.cub.2005.10.026 Moussian Bernard\n10.1101/gad.5.4.594 Treisman J\nNone Barad M\n10.1038/330583a0 Rushlow C\n10.1016/0092-8674(88)90217-6 Seeger M A\n10.1016/s0092-8674(02)01087-5 Stathopoulos Angelike\n10.1016/0168-9525(91)90456-z Govind S\n10.5483/bmbrep.2014.47.9.122 Shin Dong-Hyeon\n10.1101/gad.6.12b.2606 Skeath J B\n10.1073/pnas.012292899 Stauber Michael\nNone Berleth T\n10.1371/journal.pgen.0010057 Barker Michael S\n10.1016/j.cub.2004.07.018 Santos Ana C\nNone Roth S\n10.1093/emboj/17.23.7009 Nibu Y\nNone Nüsslein-Volhard C\n10.1186/gb-2002-3-12-research0083 Misra Sima\n10.1016/j.ydbio.2006.05.004 Anderson Jason\nNone Hariharan I K\n10.1101/gad.10.22.2922 Biehs B\nNone Arora K\n10.1016/j.devcel.2005.09.005 Stathopoulos Angelike\nNone Clifford R\n10.1093/hmg/11.21.2657 Kazemi-Esfarjani Parsa\n10.1186/s12864-017-3589-6 Chambers Michael\n10.1186/gb-2007-8-4-r67 Beltran Sergi\n10.1101/gad.8.13.1489 Mason E D\nNone Munté A\n10.1006/jtbi.1997.0523 Sanchez null\n10.1016/0955-0674(91)90115-f Lipshitz H D\n10.1016/j.cub.2013.09.055 Gavin-Smyth Jackie\n10.1038/nature13395 Kvon Evgeny Z\n10.1098/rstb.1995.0111 Gray S\n10.1073/pnas.93.18.9309 Cai H N\n10.1242/dev.099366 Upadhyai Priyanka\n10.1016/s0960-9822(00)00531-5 Gibson G\n10.1016/s0959-437x(98)80037-3 Finnerty J R\nNone Hazelrigg T\n10.1371/journal.pcbi.1000652 Kim Jaebum\n10.1016/j.ydbio.2005.12.014 Scuderi Anne\nNone Ray R P\nNone Chang T\n10.1007/bf00357688 Terol J\n10.1073/pnas.93.16.8479 Falciani F\n10.1093/nar/21.22.5041 Cavarec L\nNone Panzer S\n10.1101/gad.10.9.1131 Pritchard D K\n10.1101/gad.5.2.254 Zuo P\nNone Han K\n10.1016/0012-1606(89)90187-5 Merrill V K\n10.1016/0092-8674(89)90580-1 Han K\n10.1101/gad.7.3.491 Han K\n10.1016/s1534-5807(01)00057-0 Podos S D\nNone George H\n10.1016/0925-4773(95)00349-5 Tatei K\n10.1073/pnas.96.19.10679 Brickman J M\n10.1006/dbio.1997.8625 Saha M S\nNone Müller H A\n10.1038/nrg1726 Pearson Joseph C\n10.1016/j.cell.2010.09.047 Neely G Gregory\n10.1093/nar/gni184 Hoskins Roger A\n10.1101/gad.3.9.1437 Manseau L J\n10.1093/emboj/20.20.5725 Hasson P\n10.1016/0092-8674(89)90424-8 Biggin M D\n10.1534/g3.116.029231 Lepennetier Gildas\n10.1101/gad.5.10.1892 Pan D J\n10.1101/gad.5.3.341 Parkhurst S M\n10.1101/gad.3.10.1518 Doyle H J\n10.1101/gad.1.10.1268 Rushlow C\n10.1038/emboj.2010.202 Tipping Marla\n10.1002/(SICI)1521-1878(200003)22:3<305::AID-BIES13>3.0.CO;2-R Wilkins A S\n10.1002/bies.950121107 White R J\n10.1016/0092-8674(90)90492-w Hayashi S\n10.1016/0092-8674(92)90471-n McGinnis W\n10.1016/0167-4781(94)00234-t Courey A J\n10.1016/0168-9525(90)90001-m Laufer E\n10.1016/0168-9525(91)90273-s Budd P S\n10.1016/0304-419x(92)90008-m Read D\n10.1016/s0065-2660(08)60028-0 Rushlow C\n10.1016/s0065-2660(08)60029-2 Kaufman T C\n10.1016/s0092-8674(00)80126-9 Miklos G L\n10.1016/s0092-8674(00)80250-0 Bier E\n10.1016/s0092-8674(00)81682-7 Fraser S E\n10.1016/s0960-9822(01)00007-0 Brown S\n10.1038/35056022 Casci T\n10.1073/pnas.032685999 Michelson Alan M\n10.1093/emboj/17.23.6769 Shilo B Z\n10.1101/gad.939601 Courey A J\n10.1101/gad.976502 Barolo Scott\n10.1146/annurev.bi.63.070194.002415 Gehring W J\nNone Denell R\nNone Zhou J\n"
    }
   ],
   "source": [
    "for gene in query.results(\"list\"):\n",
    "    print(gene[1],gene[2])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you want to print only those rows where publications.doi is not None then you can add an if condition as shown below. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "10.1128/mcb.11.10.4909\n10.1016/s0378-1119(03)00493-1\n10.1016/j.yexcr.2013.10.010\n10.1006/jtbi.1996.0328\n10.1016/0092-8674(91)90651-e\n10.1101/gad.5.10.1881\n10.1038/332858a0\n10.1038/35101500\n10.1016/j.cub.2005.03.022\n10.1242/dev.001065\n10.1016/s0959-437x(98)80036-1\n10.1002/1521-1878(200102)23:2<125::AID-BIES1019>3.0.CO;2-C\n10.1002/bies.950170904\n10.1073/pnas.97.9.4442\n10.1111/j.1096-0031.1996.tb00193.x\n10.1016/0092-8674(92)90595-4\n10.1016/s1534-5807(03)00338-1\n10.1016/s0960-9822(95)00112-6\n10.1016/0092-8674(92)90596-5\n10.1007/s11060-016-2326-3\n10.1242/dev.067926\n10.1038/35068578\n10.1016/j.tig.2003.10.009\n10.1371/journal.pgen.1002769\n10.1016/s0378-1119(01)00485-1\n10.1101/gad.215459.113\n10.1016/0012-1606(92)90254-e\n10.1093/gbe/evr061\n10.1007/s00438-006-0187-8\n10.1038/ncomms8102\n10.1016/s0959-437x(02)00345-3\n10.1101/gr.104018.109\n10.1371/journal.pbio.1000456\n10.1016/j.celrep.2019.03.071\n10.1101/gr.178701\n10.1016/s0959-437x(02)00314-3\n10.1371/journal.pcbi.1001020\n10.1038/ng1908\n10.1080/19420889.2018.1447743\n10.1074/jbc.271.7.3930\n10.1016/j.semcdb.2014.04.036\n10.1016/j.devcel.2011.05.009\n10.7554/eLife.04837\n10.1242/dev.019802\n10.1128/mcb.18.11.6584\n10.1038/nature07214\n10.1101/gad.11.15.1963\n10.1016/s0959-437x(96)80062-1\n10.1016/s0960-9822(95)00001-7\n10.1016/j.ydbio.2014.02.007\n10.1242/dev.00161\n10.1242/dev.00548\n10.1073/pnas.1707467114\n10.1073/pnas.1713930115\n10.1126/science.aam5339\n10.1006/dbio.1994.1306\n10.1534/genetics.116.186783\n10.1371/journal.pbio.0050117\n10.1038/sj.emboj.7601532\n10.1016/0012-1606(84)90182-9\n10.1016/0959-437x(93)90090-c\n10.1006/jtbi.1995.0175\n10.1101/gad.6.9.1654\n10.1101/gr.6490707\n10.1146/annurev-ento-120811-153601\n10.1016/s0965-1748(02)00089-9\n10.1093/nar/28.2.454\n10.1016/s0959-437x(03)00017-0\n10.1038/380037a0\n10.1371/journal.pgen.1006868\n10.1146/annurev.genet.35.102401.090832\n10.1101/gad.1509607\n10.1038/ncomms10115\n10.1016/0955-0674(93)90017-k\n10.1242/dev.01722\n10.1371/journal.pone.0030610\n10.1101/gad.188052.112\n10.1006/dbio.2002.0652\n10.1007/s00412-015-0543-8\n10.1101/gr.209486.116\n10.1016/j.ydbio.2006.08.027\n10.1002/1097-010X(20001215)288:4<345::AID-JEZ7>3.0.CO;2-Y\n10.1098/rstb.1995.0119\n10.1101/gr.3468605\n10.1016/0168-9525(94)90132-5\n10.1371/journal.pbio.0060027\n10.1038/323076a0\n10.1016/j.cub.2010.01.040\n10.1002/bies.20285\n10.1101/gad.11.15.1949\n10.1534/genetics.112.144915\n10.1371/journal.pcbi.1004159\n10.1007/s00438-010-0591-y\n10.1038/nature07388\n10.1038/nature03318\n10.1371/journal.pgen.1002086\n10.1371/journal.pgen.1002339\n10.1016/j.ymeth.2014.01.003\n10.1016/s1097-2765(03)00500-8\n10.1016/s0955-0674(96)80010-x\n10.1186/1471-2148-6-106\n10.1016/s0955-0674(96)80067-6\n10.1016/s0959-437x(00)00130-1\n10.1093/nar/gkr672\n10.7554/eLife.00861\n10.1016/j.ydbio.2010.06.026\n10.1042/bj3310001\n10.1534/genetics.109.110122\n10.1016/s0968-0004(00)01597-8\n10.1073/pnas.012591199\n10.1016/j.gde.2004.07.004\n10.1186/1471-2105-5-202\n10.1006/dbio.1997.8689\n10.1242/dev.02689\n10.1242/dev.02251\n10.1016/j.cois.2016.01.009\n10.1016/s0960-9822(03)00576-1\n10.1242/dev.02238\n10.1101/gad.494808\n10.1093/nar/gkp619\n10.1038/nrg1724\n10.1016/S0022-2836(05)80201-3\n10.1007/BF00375954\n10.1371/journal.pgen.1005969\n10.1186/1752-0509-6-31\n10.1371/journal.pone.0001115\n10.1093/emboj/19.17.4463\n10.1038/nrg947\n10.1038/nature15545\n10.1016/s0012-1606(03)00353-1\n10.1016/0959-437x(93)90092-4\n10.1242/dev.079772\n10.1101/gad.8.21.2602\n10.1016/s0960-9822(06)00104-7\n10.1016/j.devcel.2011.12.007\n10.1016/j.ydbio.2013.02.010\n10.1093/molbev/msu286\n10.1371/journal.pgen.1000178\n10.1007/s10709-006-0031-4\n10.1038/nmeth.1548\n10.1016/j.cub.2003.12.036\n10.1371/journal.pgen.1002905\n10.1016/j.ygeno.2008.09.006\n10.1093/molbev/msw132\n10.1016/s0959-437x(02)00344-1\n10.1093/emboj/20.13.3298\n10.1091/mbc.e06-09-0790\n10.1101/gad.1350705\n10.1126/science.284.5414.606\n10.1371/journal.pgen.1007631\n10.1038/nature08799\n10.1016/s0960-9822(03)00472-x\n10.1093/nar/25.4.794\n10.1101/gad.11.22.2952\n10.1128/mcb.22.14.5089-5099.2002\n10.1101/gad.9.24.3177\n10.1016/s0378-1119(00)00161-x\n10.1038/sj.onc.1203223\n10.1016/j.cub.2006.05.050\n10.1016/s0960-9822(99)80381-9\n10.1016/0955-0674(91)90112-c\n10.1016/s0012-1606(03)00448-2\n10.1073/pnas.96.7.3786\n10.1016/0925-4773(95)00439-8\n10.7554/eLife.12068\n10.1073/pnas.89.17.7861\n10.1371/journal.pone.0098585\n10.1242/dev.02373\n10.1534/genetics.166.3.1323\n10.1534/genetics.105.041327\n10.1016/0012-1606(89)90143-7\n10.1002/bies.10180\n10.1093/emboj/18.3.605\n10.1128/mcb.15.7.3627\n10.1242/dev.02643\n10.1016/j.ymeth.2014.03.016\n10.1016/j.smim.2014.05.003\n10.7554/eLife.34594\n10.1073/pnas.93.18.9322\n10.1016/j.tig.2006.12.001\n10.1073/pnas.90.1.143\n10.1002/bies.950140302\n10.1101/gad.8.11.1247\n10.1146/annurev.ge.29.120195.002103\n10.1371/journal.pgen.1002266\n10.1101/gad.861401\n10.1073/pnas.97.9.4438\n10.1371/journal.pone.0010820\n10.1007/pl00006577\n10.1126/science.287.5461.2185\n10.1016/j.jtbi.2008.01.027\n10.1371/journal.pone.0029172\n10.1128/mcb.10.3.872\n10.1073/pnas.0408031102\n10.1016/j.cub.2005.10.026\n10.1101/gad.5.4.594\n10.1038/330583a0\n10.1016/0092-8674(88)90217-6\n10.1016/s0092-8674(02)01087-5\n10.1016/0168-9525(91)90456-z\n10.5483/bmbrep.2014.47.9.122\n10.1101/gad.6.12b.2606\n10.1073/pnas.012292899\n10.1371/journal.pgen.0010057\n10.1016/j.cub.2004.07.018\n10.1093/emboj/17.23.7009\n10.1186/gb-2002-3-12-research0083\n10.1016/j.ydbio.2006.05.004\n10.1101/gad.10.22.2922\n10.1016/j.devcel.2005.09.005\n10.1093/hmg/11.21.2657\n10.1186/s12864-017-3589-6\n10.1186/gb-2007-8-4-r67\n10.1101/gad.8.13.1489\n10.1006/jtbi.1997.0523\n10.1016/0955-0674(91)90115-f\n10.1016/j.cub.2013.09.055\n10.1038/nature13395\n10.1098/rstb.1995.0111\n10.1073/pnas.93.18.9309\n10.1242/dev.099366\n10.1016/s0960-9822(00)00531-5\n10.1016/s0959-437x(98)80037-3\n10.1371/journal.pcbi.1000652\n10.1016/j.ydbio.2005.12.014\n10.1007/bf00357688\n10.1073/pnas.93.16.8479\n10.1093/nar/21.22.5041\n10.1101/gad.10.9.1131\n10.1101/gad.5.2.254\n10.1016/0012-1606(89)90187-5\n10.1016/0092-8674(89)90580-1\n10.1101/gad.7.3.491\n10.1016/s1534-5807(01)00057-0\n10.1016/0925-4773(95)00349-5\n10.1073/pnas.96.19.10679\n10.1006/dbio.1997.8625\n10.1038/nrg1726\n10.1016/j.cell.2010.09.047\n10.1093/nar/gni184\n10.1101/gad.3.9.1437\n10.1093/emboj/20.20.5725\n10.1016/0092-8674(89)90424-8\n10.1534/g3.116.029231\n10.1101/gad.5.10.1892\n10.1101/gad.5.3.341\n10.1101/gad.3.10.1518\n10.1101/gad.1.10.1268\n10.1038/emboj.2010.202\n10.1002/(SICI)1521-1878(200003)22:3<305::AID-BIES13>3.0.CO;2-R\n10.1002/bies.950121107\n10.1016/0092-8674(90)90492-w\n10.1016/0092-8674(92)90471-n\n10.1016/0167-4781(94)00234-t\n10.1016/0168-9525(90)90001-m\n10.1016/0168-9525(91)90273-s\n10.1016/0304-419x(92)90008-m\n10.1016/s0065-2660(08)60028-0\n10.1016/s0065-2660(08)60029-2\n10.1016/s0092-8674(00)80126-9\n10.1016/s0092-8674(00)80250-0\n10.1016/s0092-8674(00)81682-7\n10.1016/s0960-9822(01)00007-0\n10.1038/35056022\n10.1073/pnas.032685999\n10.1093/emboj/17.23.6769\n10.1101/gad.939601\n10.1101/gad.976502\n10.1146/annurev.bi.63.070194.002415\n"
    }
   ],
   "source": [
    "for row in query.results(row=\"list\"):\n",
    "    if row[1]!=None:\n",
    "        print(row[1])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You can pass two more parameters while passing query.results(). These are start and size. Start represents the row number that you want to start processing from. By default this is set to 0 (first row). Size represents the number of rows that you want to print. Lets say we want to print rows 10 and 11 only. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "Gene: publications.abstractText='Although genes have specific phenotypic consequences in a given species, this functional relationship can clearly change during the course of evolution. Many cases of evolutionary dissociations between homologous genes and homologous morphological features are now known. These dissociations have interesting and important implications for understanding the genetic basis for evolutionary change in morphology.' publications.doi='10.1016/s0959-437x(98)80036-1' publications.firstAuthor='Wray G A' publications.id=1008116 publications.issue='6' publications.journal='Curr. Opin. Genet. Dev.' publications.month='Dec' publications.pages='675-80' publications.pubMedId='9914205' publications.title='When is homology not homology?' publications.volume='8' publications.year=1998\nGene: publications.abstractText='Although genes involved in common developmental programs are usually scattered throughout the metazoan genome, there are some important examples of functionally interconnected regulatory genes that display close physical linkage. In particular the homeotic genes, which determine the identities of body parts, are clustered in the Hox complexes and clustering is thought to be crucial for the proper execution of their developmental programs. Here we describe the organization and functional properties of a more recently identified cluster of six homeobox genes at 93DE on the third chromosome of Drosophila. These genes, which include tinman, bagpipe, ladybird early, ladybird late, C15, and slouch, all participate in mesodermal patterning and differentiation programs and show multiple regulatory interactions among each other. We propose that their clustering, through unknown mechanisms, is functionally significant and discuss the similarities and differences between the 93DE homeobox gene cluster and the Hox complexes.' publications.doi='10.1002/1521-1878(200102)23:2<125::AID-BIES1019>3.0.CO;2-C' publications.firstAuthor='Jagla K' publications.id=1008070 publications.issue='2' publications.journal='Bioessays' publications.month='Feb' publications.pages='125-33' publications.pubMedId='11169585' publications.title='A cluster of Drosophila homeobox genes involved in mesoderm differentiation programs.' publications.volume='23' publications.year=2001\n"
    }
   ],
   "source": [
    "for row in query.results(row=\"rr\",size=2,start=10):\n",
    "    print(row)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you prefer dealing with lists of strings, i.e. csv and tsv objects you can use them too. First, you need to import the csv library. If you want to read your results in a CSV format, you should create a csv.reader object. This has been shown below. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "csv_reader = csv.reader(query.results(row=\"csv\"), delimiter=\",\", quotechar='\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "tags": [
     "outputPrepend",
     "outputPrepend",
     "outputPrepend"
    ]
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "onverts angiotensin I (A I) into angiotensin II (A II). A II functions as a potent vasoconstrictor by triggering a G-coupled receptor system in the smooth muscles that line blood vessels. Drosophila Race is composed of 615 amino acid residues, and shares extensive sequence identity with mammalian ACE over its entire length (over 42% overall identity and greater than 60% similarity). Evidence is presented that Race might correspond to a target of the homeobox regulatory gene, zerknullt (zen). Soon after zen expression is restricted to the dorsal-most regions of the embryonic ectoderm, Race is activated in a coincident pattern and becomes associated with the amnioserosa during germ band elongation, shortening and heart morphogenesis. After germ band elongation, Race is also expressed in both the anterior and posterior midgut, where it persists throughout embryogenesis. Race expression is lost from the dorsal ectoderm in either zen- or dpp- mutants, although gut expression is unaffected. P-transformation assays and genetic complementation tests suggest that Race corresponds to a previously characterized lethal complementation group, 1(2)34Eb. Mutants die during larval/pupal development, and transheterozygotes for two different lethal alleles exhibit male sterility. We propose that Race might play a role in the contractions of the heart, gut, or testes and also suggest that Hox genes might be important for coordinating both developmental and physiological processes.\nWe show that the Drosophila protein DSP1, an HMG-1/2-like protein, binds DNA highly cooperatively with three members of the Rel family of transcriptional regulators (NF-kappaB, the p50 subunit of NF-kappaB, and the Rel domain of Dorsal). This cooperativity is apparent with DNA molecules bearing consensus Rel-protein-binding sites and is unaffected by the presence of a negative regulatory element, a sequence previously proposed to be important for mediating repression by these Rel proteins. The cooperativity observed in these DNA-binding assays is paralleled by interactions between protein pairs in the absence of DNA. We also show that in HeLa cells, as assayed by transient transfection, expression of DSP1 increases activation by Dorsal from the twist promoter and inhibits that activation from the zen promoter, consistent with the previously proposed idea that DSP1 can affect the action of Dorsal in a promoter-specific fashion.\nWhile the role of the notochord and floor plate in patterning the dorsal-ventral (D/V) axis of the neural tube is clearly established, relatively little is known about the earliest stages of D/V regionalization. In an effort to examine more closely the initial, preneural plate stages of regionalization along the prospective D/V neural axis, we have performed a series of explant experiments employing xHB9, a novel marker of the motor neuron region in Xenopus. Using tissue recombinants and Keller explants we show that direct mesodermal contact is both necessary and sufficient for the initial induction of xHB9 in the motor neuron region. We also show that presumptive neural plate explants removed as early as midgastrulation and cultured in isolation are already specified to express xHB9 but do so in an inappropriate spatial pattern while identical explants are specified to express the floor plate marker vhh-1 with correct spatial patterning. Our data suggest that, in addition to floor plate signaling, continued interactions with the underlying mesoderm through neural tube stages are essential for proper spatial patterning of the motor neuron region.\nWingless signaling plays a central role during epidermal patterning in Drosophila. We have analyzed zygotic requirements for Wingless signaling in the embryonic ectoderm by generating synthetic deficiencies that uncover more than 99% of the genome. We found no genes required for initial wingless expression, other than previously identified segmentation genes. In contrast, maintenance of wingless expression shows a high degree of zygotic transcriptional requirements. Besides known genes, we have identified at least two additional genomic regions containing new genes involved in Wingless maintenance. We also assayed for the zygotic requirements for Wingless response and found that no single genomic region was required for the cytoplasmic accumulation of Armadillo in the receiving cells. Surprisingly, embryos homozygously deleted for the candidate Wingless receptor, Dfrizzled2, showed a normal Wingless response. However, the Armadillo response to Wingless was strongly reduced in double mutants of both known members of the frizzled family in Drosophila, frizzled and Dfrizzled2. Based on their expression pattern during embryogenesis, different Frizzled receptors may play unique but overlapping roles in development. In particular, we suggest that Frizzled and Dfrizzled2 are both required for Wingless autoregulation, but might be dispensable for late Engrailed maintenance. While Wingless signaling in embryos mutant for frizzled and Dfrizzled2 is affected, Wingless protein is still internalized into cells adjacent to wingless-expressing cells. Incorporation of Wingless protein may therefore involve cell surface molecules in addition to the genetically defined signaling receptors of the frizzled family.\nWith their power to shape animal morphology, few genes have captured the imagination of biologists as the evolutionarily conserved members of the Hox clusters have done. Recent research has provided new insight into how Hox proteins cause morphological diversity at the organismal and evolutionary levels. Furthermore, an expanding collection of sequences that are directly regulated by Hox proteins provides information on the specificity of target-gene activation, which might allow the successful prediction of novel Hox-response genes. Finally, the recent discovery of microRNA genes within the Hox gene clusters indicates yet another level of control by Hox genes in development and evolution.\nWorldwide, acute, and chronic pain affects 20% of the adult population and represents an enormous financial and emotional burden. Using genome-wide neuronal-specific RNAi knockdown in Drosophila, we report a global screen for an innate behavior and identify hundreds of genes implicated in heat nociception, including the α2δ family calcium channel subunit straightjacket (stj). Mice mutant for the stj ortholog CACNA2D3 (α2δ3) also exhibit impaired behavioral heat pain sensitivity. In addition, in humans, α2δ3 SNP variants associate with reduced sensitivity to acute noxious heat and chronic back pain. Functional imaging in α2δ3 mutant mice revealed impaired transmission of thermal pain-evoked signals from the thalamus to higher-order pain centers. Intriguingly, in α2δ3 mutant mice, thermal pain and tactile stimulation triggered strong cross-activation, or synesthesia, of brain regions involved in vision, olfaction, and hearing.\ncDNA cloning is a central technology in molecular biology. cDNA sequences are used to determine mRNA transcript structures, including splice junctions, open reading frames (ORFs) and 5'- and 3'-untranslated regions (UTRs). cDNA clones are valuable reagents for functional studies of genes and proteins. Expressed Sequence Tag (EST) sequencing is the method of choice for recovering cDNAs representing many of the transcripts encoded in a eukaryotic genome. However, EST sequencing samples a cDNA library at random, and it recovers transcripts with low expression levels inefficiently. We describe a PCR-based method for directed screening of plasmid cDNA libraries. We demonstrate its utility in a screen of libraries used in our Drosophila EST projects for 153 transcription factor genes that were not represented by full-length cDNA clones in our Drosophila Gene Collection. We recovered high-quality, full-length cDNAs for 72 genes and variously compromised clones for an additional 32 genes. The method can be used at any scale, from the isolation of cDNA clones for a particular gene of interest, to the improvement of large gene collections in model organisms and the human. Finally, we discuss the relative merits of directed cDNA library screening and RT-PCR approaches.\ncappuccino and spire are unique Drosophila maternal-effect loci that participate in pattern formation in both the anteroposterior and dorsoventral axes of the early embryo. Mutant females produce embryos lacking pole cells, polar granules, and normal abdominal segmentation. They share these defects with the posterior group of maternal-effect genes. Although embryos are defective in abdominal segmentation, in double mutant combinations with Bicaudal D, abdominal segments can be formed in the anterior half of the egg. This indicates that embryos produced by mutant females contain the 'posterior determinant' required for abdominal segmentation (Nüsslein-Volhard et al. 1987) and suggests that the wild-type gene products are not required for production of the posterior determinant but, rather, for its localization or stabilization. The vasa protein, a component of polar granules, is not localized at the posterior pole of mutant egg chambers or embryos, providing additional support for the hypothesis that localization to or stabilization of substances at the posterior pole of the egg chamber is defective in mutant females. Females mutant for the strongest alleles also produce dorsalized embryos. Phenotypic analysis reveals that these dorsalized embryos also have abdominal segmentation defects. The mutant phenotypes can be ordered in a series of increasing severity. Pole cell formation is most sensitive to loss of functional gene products, followed by abdominal segmentation, whereas normal dorsoventral patterning is the least sensitive to loss of functional gene products. In addition, mutant females contain egg chambers that appear to be dorsalized, resulting in the production of eggs with dorsalized eggshells. Germ-line mosaics indicate that cappuccino and spire are required in the oocyte-nurse cell complex. This suggests that the eggshell phenotype results from altered pattern in the underlying germ cell. Also, we defined the epistatic relationships between several early patterning loci, on the basis of an analysis of the eggs and embryos produced by females doubly mutant for cappuccino or spire and other loci that affect the pattern of both the egg and the embryo. On the basis of our current knowledge of the genes involved in this process, we formulated a working model for the early steps in dorsoventral patterning.\ndecapentaplegic (dpp) encodes a Drosophila transforming growth factor-beta homologue that functions as a morphogen in the developing embryo and in adult appendage formation. In the wing imaginal disc, a Dpp gradient governs patterning along the anteroposterior axis by inducing regional expression of diverse genes in a concentration-dependent manner. Recent studies show that responses to graded Dpp activity also require an input from a complementary and opposing gradient of Brinker (Brk), a transcriptional repressor protein encoded by a Dpp target gene. Here we show that Brk harbours a functional and transferable repression domain, through which it recruits the corepressors Groucho and CtBP. By analysing transcriptional outcomes arising from the genetic removal of these corepressors, and by ectopically expressing Brk variants in the embryo, we demonstrate that these corepressors are alternatively used by Brk for repressing some Dpp-responsive genes, whereas for repressing other distinct target genes they are not required. Our results show that Brk utilizes multiple means to repress its endogenous target genes, allowing repression of a multitude of complex Dpp target promoters.\neven-skipped (eve) is a homeodomain-encoding gene that is a genetically defined repressor of Ultrabithorax (Ubx), fushi-tarazu (ftz), and wingless (wg). Here we report that purified eve protein represses transcription in vitro at the Ubx promoter, in a DNA binding site-dependent manner. eve protein represses transcription when bound either upstream or downstream of the RNA start site or when DNA binding sites are in either orientation. We also show that eve represses expression from the Ubx promoter in Drosophila tissue culture cells, again in a binding site-dependent manner. Deletion of eve DNA binding sites does not alter transcription in the absence of eve, and so repression is not likely to be the result of eve competitively inhibiting an activator protein from binding to the same DNA element. Instead, we propose that eve protein is probably interfering with the function of proteins bound at other locations in the promoter. The biochemical demonstration that a Drosophila homeodomain protein can directly regulate RNA synthesis strengthens the view that this class of regulators act as transcription factors to control development.\nmRNA-associated processes and gene structure in eukaryotes are typically treated as separate research subjects. Here, we bridge this separation and leverage the extensive multidisciplinary work on Drosophila melanogaster to examine the roles that capping, splicing, cleavage/polyadenylation, and telescripting (i.e, the protection of nascent transcripts from premature cleavage/polyadenylation by the splicing factor U1) might play in shaping exon-intron architecture in protein-coding genes. Our findings suggest that the distance between subsequent internal 5' splice sites (5'ss) in Drosophila genes is constrained such that telescripting effects are maximized, in theory, and thus nascent transcripts are less vulnerable to premature termination. Exceptionally weak 5'ss and constraints on intron-exon size at the gene 5' end also indicate that capping might enhance the recruitment of U1 and, in turn, promote telescripting at this location. Finally, a positive correlation between last exon length and last 5'ss strength suggests that optimal donor splice sites in the proximity of the pre-mRNA tail may inhibit the processing of downstream polyadenylation signals more than weak donor splice sites do. These findings corroborate and build upon previous experimental and computational studies on Drosophila genes. They support the possibility, hitherto scantly explored, that mRNA-associated processes impose significant constraints on the evolution of eukaryotic gene structure.\ntwist is one of the earliest expressed zygotically active genes required for dorsal-ventral pattern formation in the Drosophila embryo. Genetic studies suggest that this gene is activated in the ventral part of the blastoderm by maternally expressed dorsal gene product. Using P-element-mediated germ-line transformation, we have mapped a small (260 bp) dorsal-dependent ventral activator region (VAR) in the 5'-flanking region of the twist promoter that can direct the early ventral expression of a heterologous promoter. The VAR contains binding sites for a number of proteins present in extracts of Drosophila embryos. One of these sites bears homology to known binding sites for the dorsal transcription factor and is specifically bound by bacterially expressed dorsal protein. Furthermore, a 37-bp deletion that removes the dorsal-binding sequences abolishes the ventral-specific activity of the twist promoter constructs. Our data also show that additional sequences within the VAR are required to render the dorsal-binding sites functional. Finally, reverse genetic and biochemical data suggest that the transcription factor, encoded by the zeste gene may help control the overall level, if not the pattern, of twist expression.\nwimp is a dominant maternal-effect mutation that interacts with a specific subset of early-acting maternal and zygotic Drosophila genes. We show that wimp is a change-of-function mutation, allelic to mutations of the 140-kD subunit of RNA polymerase, which causes reduced transcription of interacting genes. Loci that do not interact with wimp are expressed at normal levels. We discuss these results in terms of specific interactions between transcription factors and RNA polymerase. Embryos from wimp mothers show unaltered fate maps and develop normally, despite the reduction of transcript levels at least twofold. We suggest that spatial cues are determined by a balance of segmentation gene products rather than their absolute concentrations. We also demonstrate powerful genetic screens for otherwise undetected loci required for segmentation, sex determination, and other early functions.\nzerknüllt (zen) is one of approximately 10 zygotically active genes that control the differentiation of the dorsal-ventral (D/V) pattern during early embryogenesis in Drosophila. Past genetic analyses suggest that maternal factors repress the expression of zen in ventral regions, thereby restricting zen products to dorsal and dorsal-lateral regions of precellular embryos. Subsequent interactions with other zygotic D/V regulatory genes refine the zen pattern, restricting expression to the dorsal-most ectoderm. Here we describe the use of zen promoter fusions and P-element transformation to identify cis elements that are responsible for the complex spatial pattern of zen expression. The zen promoter shows a two-tier organization: Distal sequences mediate its initial response to maternal factors, whereas proximal sequences are responsible for the refinement of the pattern in older embryos. The distal regulatory element has the property of a silencer (or anti-enhancer) element and can act over a distance to repress ventral expression of a heterologous promoter. Also, we discuss evidence that proximal promoter sequences interact with factors that may be modulated by a cell-cell communication pathway.\nzerknüllt (zen) is unique among the 18 known homeo box genes in Drosophila since it is required for the differentiation of the dorsal-ventral pattern, and does not appear to be involved in the process of segmentation. Here we show that the zen region of the Antennapedia complex (ANT-C) consists of two closely linked homeo box genes, designated z1 and z2. The z1 and z2 transcription units show essentially identical patterns of expression during early development, which are consistent with the timing and sites of zen+ gene activity. The putative proteins encoded by z1 and z2 are highly divergent and are related only by virtue of homeo box homology. We have used P-element-mediated germ line transformation to show that z1 alone can provide zen+ gene function, suggesting that the z2 gene might be dispensable. The occurrence of closely linked homeo box genes that display similar patterns of expression is not unique to the zen locus. Such gene duplications might provide important clues to the evolution of the homeo box gene family in Drosophila and other organisms.\nβ-Arrestins have been implicated in the regulation of multiple signalling pathways. However, their role in organism development is not well understood. In this study, we report a new in vivo function of the Drosophila β-arrestin Kurtz (Krz) in the regulation of two distinct developmental signalling modules: MAPK ERK and NF-κB, which transmit signals from the activated receptor tyrosine kinases (RTKs) and the Toll receptor, respectively. Analysis of the expression of effectors and target genes of Toll and the RTK Torso in krz maternal mutants reveals that Krz limits the activity of both pathways in the early embryo. Protein interaction studies suggest a previously uncharacterized mechanism for ERK inhibition: Krz can directly bind and sequester an inactive form of ERK, thus preventing its activation by the upstream kinase, MEK. A simultaneous dysregulation of different signalling systems in krz mutants results in an abnormal patterning of the embryo and severe developmental defects. Our findings uncover a new in vivo function of β-arrestins and present a new mechanism of ERK inhibition by the Drosophila β-arrestin Krz.\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n"
    }
   ],
   "source": [
    "for row in csv_reader:\n",
    "    print(row[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you have used the csv library before try writing your results into a csv file using the writer class. If you have not used it before, trying going through the documentation first and then writing code on your own. It is pretty self explanatory. The documentation can be found at: https://docs.python.org/2/library/csv.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The last thing that we will look at, in this tutorial, is the summarize method. This method proves to be particularly useful when we want some basic statistics regarding a particular column. We will look at the statistics of the length of genes present in the list : List of the most enriched genes in the adult fly brain. We begin by creating a query. This is followed by adding views and the list constraint. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "query2=service.new_query()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "<intermine.query.Query at 0x11276f7c0>"
     },
     "metadata": {},
     "execution_count": 14
    }
   ],
   "source": [
    "query2.select(\"Gene.*\",\"organism.*\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": "<ListConstraint: Gene IN PL FlyAtlas_brain_top>"
     },
     "metadata": {},
     "execution_count": 15
    }
   ],
   "source": [
    "query2.add_constraint(\"Gene\",\"IN\",\"PL FlyAtlas_brain_top\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We then print out the first 10 rows of results. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "Gene: briefDescription=None cytoLocation='10A3-10A3' description=None id=1104775 length=2075 name=None primaryIdentifier='FBgn0030259' score=None scoreType=None secondaryIdentifier='CG1545' symbol='CG1545' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='11D8-11D8' description=None id=1068742 length=90456 name='radish' primaryIdentifier='FBgn0265597' score=None scoreType=None secondaryIdentifier='CG44424' symbol='rad' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='14A1-14A1' description=None id=1039501 length=26224 name='mind-meld' primaryIdentifier='FBgn0259110' score=None scoreType=None secondaryIdentifier='CG42252' symbol='mmd' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='16F3-16F5' description=None id=1058279 length=138941 name='Shaker' primaryIdentifier='FBgn0003380' score=None scoreType=None secondaryIdentifier='CG12348' symbol='Sh' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='18C2-18C3' description=None id=1074968 length=21373 name='nicotinic Acetylcholine Receptor alpha7' primaryIdentifier='FBgn0086778' score=None scoreType=None secondaryIdentifier='CG32538' symbol='nAChRalpha7' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='19A4-19A4' description=None id=1068367 length=58027 name='Dopamine 2-like receptor' primaryIdentifier='FBgn0053517' score=None scoreType=None secondaryIdentifier='CG33517' symbol='Dop2R' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='24C9-24D1' description=None id=1008553 length=88786 name='friend of echinoid' primaryIdentifier='FBgn0051774' score=None scoreType=None secondaryIdentifier='CG31774' symbol='fred' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='25B4-25B4' description=None id=1693557 length=1564 name=None primaryIdentifier='FBgn0031650' score=None scoreType=None secondaryIdentifier='CG14044' symbol='CG14044' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='30D1-30E1' description=None id=1428781 length=92934 name='nicotinic Acetylcholine Receptor alpha6' primaryIdentifier='FBgn0032151' score=None scoreType=None secondaryIdentifier='CG4128' symbol='nAChRalpha6' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\nGene: briefDescription=None cytoLocation='42D4-42D6' description=None id=1176763 length=10005 name=None primaryIdentifier='FBgn0033108' score=None scoreType=None secondaryIdentifier='CG15236' symbol='CG15236' organism.commonName='fruit fly' organism.genus='Drosophila' organism.id=1000010 organism.name='Drosophila melanogaster' organism.shortName='D. melanogaster' organism.species='melanogaster' organism.taxonId='7227'\n"
    }
   ],
   "source": [
    "for row in query2.rows(size=10):\n",
    "    print(row)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can look at the summary of the length of each gene. This contains some useful information such as the average length and the maximum and minimum length. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "{'bucket': 1.0, 'average': 31601.891891891893, 'min': 808.0, 'max': 169106.0, 'buckets': 20.0, 'count': 17.0, 'stdev': 41489.79692298}\n"
    }
   ],
   "source": [
    "print(query2.summarise(\"length\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " Also, another property that comes in handy while printing results about a query that we can print only the columns we want in that result. Suppose for now, I only want the ID of Gene's, here's what I do: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": "1104775\n1068742\n1039501\n1058279\n1074968\n1068367\n1008553\n1693557\n1428781\n1176763\n"
    }
   ],
   "source": [
    " for row in query2.rows(size=10):\n",
    "    print(row[\"Gene.id\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This brings us to the end of the fifth tutorial. The next tutorial will be about further management of results. "
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3.8.1 64-bit",
   "language": "python",
   "name": "python38164bit64f092a14f47420fadd99522ecd01b91"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1-final"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}